server/services/distance.ts

```diff
@@ -3,11 +3,38 @@ import { Location } from "../types/Location";
 
 export default class DistanceService {  
 
-    getDistanceBetweenLocations(loc1: Location, loc2: Location): number {
-        // Math.hypot returns the root of the squared values of its variables. So this is sqrt( (x2 - x1)^2 + (y2 - y1)^2 ), much shorter than the older version
+    getDistanceBetweenLocations(loc1: Location, loc2: Location) {
         return Math.hypot(loc2.x - loc1.x, loc2.y - loc1.y);
     }
 
+    getDistanceAlongLocationList(locations: Location[]) {
+        if (!locations || locations.length < 2) {
+            return 0;
+        }
+
+        let distance = 0;
+        let last = locations[0];
+
+        for (let i = 1; i < locations.length; i++) {
+            const current = locations[i];
+            distance += this.getDistanceBetweenLocations(last, current);
+            last = current;
+        }
+
+        return distance;
+    }
+
+    getDistanceSquaredBetweenLocations(loc1: Location, loc2: Location)
```

Comment:
I'm no maths wizard but is this any different than the one above on line 6?

Summoning @Tristanvds